Remove openSUSE and reduce everything to one build

thardeck opened the pull request:
At the moment the SLE12 stemcell is still in the staging registry but this should change in the nearby future with SLE15, where we should be able to use the openSUSE registry.

f0rmiga commented on .envrc:
```diff
@@ -90,12 +83,7 @@ if ! type source_up >/dev/null 2>/dev/null; then
-else
-    export FISSILE_STEMCELL=${FISSILE_STEMCELL:-${FISSILE_DOCKER_REGISTRY}${FISSILE_DOCKER_ORGANIZATION}/fissile-stemcell-sle12:$FISSILE_STEMCELL_VERSION}
-fi
+export FISSILE_STEMCELL=${FISSILE_STEMCELL:-${FISSILE_DOCKER_REGISTRY}${FISSILE_DOCKER_ORGANIZATION}/fissile-stemcell-sle12:$FISSILE_STEMCELL_VERSION}
```
Could you pull `${FISSILE_DOCKER_REGISTRY}${FISSILE_DOCKER_ORGANIZATION}/fissile-stemcell-sle12:$FISSILE_STEMCELL_VERSION` into a separate variable? Maybe `FISSILE_DEFAULT_STEMCELL`. It doesn't need to be exported. It will ease the reading.

f0rmiga commented on README.md:
```diff
@@ -124,14 +124,11 @@ a working system.
-   This allows direnv to configure the various stemcell and stack env vars. The
-   `FISSILE_DOCKER_REPOSITORY` env var will need to be set, and Docker configured
-   to login to the repository.
+   The `FISSILE_DOCKER_REPOSITORY` env var will need to be set, and Docker
```
env var -> environment variable

f0rmiga commented on container-host-files/etc/scf/config/scripts/patches/mapfs_pre_start.sh:
```diff
@@ -63,7 +63,7 @@ index 7574622..f3cba25 100644
 +        ;;
 +    *suse-*)
-+        # openSUSE / SUSE
++        # SUSE
```
I don't think this comment is necessary.

f0rmiga commented on make/bundle-dist:
```diff
@@ -11,17 +11,11 @@ GIT_ROOT="${GIT_ROOT:-$(git rev-parse --show-toplevel)}"
-    stemcell_os="sle"
-    suffix=""
-else
+if [[ ! ${FISSILE_STEMCELL} =~ ^.*fissile-stemcell-sle.*$ ]] ; then
```
Need to wrap `${FISSILE_STEMCELL}` in double quotes.

f0rmiga commented on src/scf-release/src/acceptance-tests-brain/test-scripts/017_syslog_forwarding_test.rb:
```diff
@@ -119,6 +119,7 @@ def emit_log_entries
     socat_args = "/usr/bin/socat #{SCF_LOG_PROTOCOL.upcase}-LISTEN:#{SCF_LOG_PORT},fork 'EXEC:/usr/bin/logger --socket-errors=off --stderr --tag \"\"'"
     cmd = %W(
         kubectl run #{$LOG_SERVICE_NAME}
+            --generator=run-pod/v1
```
There is an open PR for this bit already https://github.com/SUSE/scf/pull/2187/files. Since we skipped brain tests on this PR, how about removing it from here?

f0rmiga commented on tooling/lib/releases_diff.rb:
```diff
@@ -33,12 +33,7 @@ def initialize(manifest_path = nil, output = nil)
-        else
-            stack = ENV['FISSILE_LIGHT_SLE12']
-        end
+        stack = ENV["FISSILE_LIGHT_OPINIONS"]
```
I know you simply removed the condition, but the label of this variable is very misleading IMO. How about renaming it and making it a const?
```ruby
FISSILE_LIGHT_OPINIONS = ENV["FISSILE_LIGHT_OPINIONS"]
```

f0rmiga commented on Jenkinsfile:
```diff
@@ -872,8 +834,8 @@ pass = ${OBS_CREDENTIALS_PASSWORD}
-                    sh 'bin/clean-jenkins-log'
-                    sh 'container-host-files/opt/scf/bin/klog.sh -f ${jobBaseName()}-${BUILD_NUMBER}-scf'
+                    sh "bin/clean-jenkins-log"
+                    sh "container-host-files/opt/scf/bin/klog.sh -f ${jobBaseName()}-${BUILD_NUMBER}-scf"
```
This is already on develop. https://github.com/SUSE/scf/blob/develop/Jenkinsfile#L875-L876

thardeck commented:
@f0rmiga I have incorporated your changes and rebased against current develop. Please verify if the changes are like you expected.

There seems to be a deploy issue with the diego cell which was not before. If you see the issue directly feel free to fix it otherwise I will debug it tomorrow.

f0rmiga commented on .envrc:
```diff
@@ -90,13 +83,10 @@ if ! type source_up >/dev/null 2>/dev/null; then
     PWD="$(pwd)"
 fi
 
+FISSILE_DEFAULT_STEMCELL=${FISSILE_STEMCELL:-${FISSILE_DOCKER_REGISTRY}${FISSILE_DOCKER_ORGANIZATION}/fissile-stemcell-sle12:$FISSILE_STEMCELL_VERSION}
```
I don't think we want this as it is weird to access `FISSILE_STEMCELL` in the `FISSILE_DEFAULT_STEMCELL` and then export `FISSILE_STEMCELL` with the default value. The point of my previous comment was to make it easier to read, not harder. Instead, do
```
export FISSILE_STEMCELL=${FISSILE_STEMCELL:-$FISSILE_DEFAULT_STEMCELL}
```
and remove `FISSILE_STEMCELL` from the `FISSILE_DEFAULT_STEMCELL`.

Sorry, I was not very explicit the first time.

thardeck commented:
@f0rmiga I had to revert the change regarding the comment because it would break the inline patch and I do not think it is relevant enough to regenerate it.
The Jenkins run already looks good and if the changes are fine with you I will merge them all together to one commit and will merge the pr.
